undertow/WFLY-17273_Global_tracking_mode_setting_in_Undertow.adoc

```diff
@@ -0,0 +1,100 @@
+= WFLY-17273 Global tracking-mode setting in Undertow
+:author:            Lin Gao
+:email:             lgao@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+There is an element: `tracking-mode` in `session-config` section of web descriptor, which is used to specify the session tracking mode, the available value can be: `COOKIE`, `URL`, `SSL`. Example in the web.xml is like:
+
+[source,xml]
+----
+<session-config>
+    <tracking-mode>COOKIE</tracking-mode>
+    <tracking-mode>URL</tracking-mode>
+</session-config>
+----
+
+When it is not defined, undertow uses `COOKIE` and falls over to `URL` if the browser does not allow cookie for the deployed web applications.
+
+When users want to have different set up for all web applications, they need to update the web.xml in each application. It would be great if undertow subsystem supports specifying the session tracking modes globally at the servlet container level.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-17273[WFLY-17273]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1980[EAP7-1980]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+[X] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+WildFly Undertow subsystem
+
+=== Other Interested Projects
+
+N/A
+
+== Requirements
+
+Provide a way to specify global session tracking modes at servlet container level. The configuration takes effect only if there are no session tracking modes specified in the deployment. So the configuration either in `web.xml` or updated in a `ServletContextListener` implementation takes precedence, which keeps consistent behavior as what it is now.
+
+=== Hard Requirements
+
+There will be 2 ways to set up global session tracking modes in the server:
+
+- Use system property with prefix `undertow.session.tracking-modes`. It reads from system property of `undertow.session.tracking-modes.CONTAINER-NAME`, and falls over to `undertow.session.tracking-modes` if it is not defined.
```

Comment:
I don't see why we need a system property. system property are added usually when it is too late to add something to the model, for example when we need a fix to a bug urgently and the fix cannot wait for the whole RFE process. Still, it is the goal to eventually add those system properties to the model and deprecate those in the code.

Reply:
There is a EAPSUP issue linked to https://issues.redhat.com/browse/WFLY-17273, I thought there is a request back porting to EAP 7.4, if that is not the case, I think we can avoid the system property approach.

@bmaxwell would you please comment if we need this back porting to EAP 7.4 ?

Reply:
It is also possible to use system property for EAP 7.4 in case it is needed, and use attribute only upstream. :)

Reply:
Thanks, what do you mean by changes in undertow ? I think it can be done in wildfly undertow subsystem directly.  Please take a look at the section of the implementation plan below. maybe there are anything I missed ?